spec/system/admin/order_cycles/edit_spec.rb

```diff
@@ -0,0 +1,166 @@
+# frozen_string_literal: true
+
+require 'system_helper'
+
+RSpec.describe '
+    As an administrator
+    I want to edit a specific order cycle
+' do
+  include AdminHelper
+  include AuthenticationHelper
+  include WebHelper
+
+  let(:order_cycle_attrs) {
+    { orders_open_at: "2024-03-01 08:00", orders_close_at: "2024-03-20 20:00", }
+  }
+  describe 'simple order cycle' do
+    let(:coordinator) { create(:distributor_enterprise, sells: 'own') }
+    let(:order_cycle) {
+      create(:simple_order_cycle, coordinator:, **order_cycle_attrs, suppliers: [coordinator],
+                                  distributors: [coordinator])
+    }
+
+    context 'with attached order(s)' do
+      let!(:order) { create(:order, order_cycle: ) }
+
+      it "shows warning modal when datetime field values change" do
+        login_as_admin
+        visit edit_admin_order_cycle_path(order_cycle)
+
+        # change non-date range field
+        fill_in 'order_cycle_name', with: "Order cycle name updated"
+        # Set Ready for value to enable save button
+        fill_in 'order_cycle_outgoing_exchange_0_pickup_time', with: 'pickup time'
+
+        expect(page).to have_content('You have unsaved changes')
+        click_button 'Save'
+
+        expect(page).not_to have_content "Orders are linked to this order cycle"
+        expect(page).to have_field 'order_cycle_name', with: "Order cycle name updated"
+        expect(page).to have_content('Your order cycle has been updated.')
+
+        # change date range field value
+        find('#order_cycle_orders_close_at').click
+        within(".flatpickr-calendar.open") do
+          expect(page).to have_selector '.shortcut-buttons-flatpickr-buttons'
+          select_datetime_from_datepicker Time.zone.parse("2024-03-30 00:00")
+          find("button", text: "Close").click
+        end
```

Comment:
I'm surprised to see that we don't have a helper method to handle this, it seems like a common action.